CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+get_filename_component(MESOS_ROOT ${MESOS_ROOT} ABSOLUTE)
+
+message("MESOS_ROOT is ${MESOS_ROOT}")
+
+# CMAKE MODULE SETUP.
+#####################
+# Paths that are searched when `include(...)` is called.
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/cmake)
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/3rdparty/cmake)
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/src/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+# Macros.
+include(CTest)
+include(Versions)
+include(Dependencies)
+
+# Set the default standard to C++11 for all targets.
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# Do not use, for example, `-std=gnu++11`.
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES MSVC)
+  # There is corresponding TODO in mesos as well.
+  #
+  # TODO(andschwa): Switch to `/W4` and re-enable possible-loss-of-data warnings.
+  #
+  # The last two warnings are disabled (well, put into `/W4`) because
+  # there is no easy equivalent to enable them for GCC/Clang without
+  # also fixing all the warnings from `-Wconversion`.
+  add_compile_options(
+    # Like `-Wall`; `/W4` is more like `-Wall -Wextra`.
+    /W3
+    # Disable permissiveness.
+    /permissive-
+    # C4244 is a possible loss of data warning for integer conversions.
+    /w44244
+    # C4267 is a possible loss of data warning when converting from `size_t`.
+    /w44267)
+endif ()
+
+if (WIN32)
+  # COFF/PE and friends are somewhat limited in the number of sections they
+  # allow for an object file. We use this to avoid those problems.
+  add_compile_options(/bigobj /vd2)
+
+  # Fix Warning C4530: C++ exception handler used, but unwind semantics are not
+  # enabled.
+  add_compile_options(/EHsc)
+
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+  # Enable multi-threaded compilation for `cl.exe`.
+  add_compile_options(/MP)
+
+  # Force use of Unicode C and C++ Windows APIs.
+  add_definitions(-DUNICODE -D_UNICODE -DNOGDI -DNOMINMAX -DGOOGLE_GLOG_DLL_DECL=)
+
+  # Convenience flags to simplify Windows support in C++ source; used to
+  # `#ifdef` out some platform-specific parts of Mesos.  We choose to define
+  # a new flag rather than using an existing flag (`_WIN32`) because we want
+  # to give the build system fine-grained control over what code is #ifdef'd
+  # out in the future.  Using only flags defined by our build system to control
+  # this logic is the clearest and most stable way of accomplishing this.
+  add_definitions(-D__WINDOWS__)
+endif ()
+
+add_subdirectory(dockercfg)
+add_subdirectory(journald)
+add_subdirectory(logsink)
+
+# FIXME(akornatskyy): due to import path in 'overlay/messages.proto'
+# NOTE: according to [1], overlay module is not required for Windows
+# [1] https://github.com/mesosphere/dcos-enterprise/tree/master/packages/mesos/extra/windows-ee.patches
+#add_subdirectory(overlay)
+
+
+# FIXME(akornatskyy): due to limitation of protobuf generation process
+# and include in source
```

Comment:
No, it is not... this has no relation to cmake specifics, the issue is with source code layout and relative paths which incorrectly resolve. I have a task in backlog so that is not an year to get fixed. As part of port to cmake linux, it would require refactor the code a bit to fix.